### PR TITLE
Enhance Slot-manager with additional informations and a method

### PR DESCRIPTION
**Pull request by Apollon77**
# Pull Request Template

## PR Checklist

- [x] I have run `npm test` locally and all tests are passing.
- [x] I have added/updated tests for any new behavior.
- [ ] If this is a significant change, an issue has already been created where the problem / solution was discussed: [N/A, or add link to issue here]

## PR Description
This PR mainly provides additional information on slot filling:
* a new property in result.slotFill states which slot/entity was filled in the current/latest run
* a new property in the entities called isSlotFillingFallback flags an entity by filled by the defined fallback logic (when trying to fill a specific slow but the given answer do not match to a property then tze full answer is filled, but in fact it is a "special" way to fill the slot. So a developer can not check and handle this

Additionally the slot-manager geta a new method to get the info if a intenthas entities defined.

### CODE REVIEW

**Comment by Apollon77** [comment]
@eric-lara Ready for review

PS: I think that the microsoft builtin test is the reason for the issues together with a memomry isue in nodejs 16 I heared of ... that test is very huge and can lead to mem isues

**Comment by ericzon** [comment]
I saw the issue you heard about v16 (https://github.com/facebook/jest/issues/11956). Running only that job should pass, but it's annoying.

File: packages/slot/src/slot-manager.js [hunk 1]
```diff
@@ -121,6 +121,16 @@ class SlotManager {
     return Object.keys(this.intents[intent]);
   }
 
+  /**
+   * Given an intent return the information if the intent had entities defined
+   *
+   * @param {String} intent Name of the intent.
+   * @returns {boolean} true if intent has defined entities, else false
+   */
+  intentHasEntities(intent) {
```
Comment (ericzon) [inline packages/slot/src/slot-manager.js:130]: maybe better some method name starting by verb like hasIntentEntities to keep consistence with other method names

Comment (Apollon77) [reply packages/slot/src/slot-manager.js]: done

**Comment by Apollon77** [comment]
@ericzon All PRs ready for review

**Review by ericzon — APPROVED** [review]

**Pull request merged**